### ISSUE TITLE
Enable experimental frontend server

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/webdev#865]
This is a meta issue tracking the frontend server integration/experimentation work.

- [ ] Enable initial serve/build behind a flag
- [ ] Enable incremental rebuilds
- [ ] Add build_runner integration for codegen
- [ ] Enable debugging with dwds

**Comment by jakemac53** [comment]
cc @grouma as I assume me and you will be working on this as time allows

**Comment by jakemac53** [comment]
I have started working on this in this branch https://github.com/dart-lang/webdev/tree/fe-server-integration and also sent out a PR to get frontend_server built with the SDK https://dart-review.googlesource.com/c/sdk/+/130134

**Comment by jakemac53** [comment]
cc @jonahwilliams as well as I have noticed some things we will likely want to change especially in terms of file naming (absolute uris in particular, but also the chosen file extensions).

Any changes will need to be coordinated with flutter_tool.

**Comment by jakemac53** [comment]
Fwiw the initial build timings based on my prototype are looking good here.

I have compiled the angular_components_example app as well as one of our largest internal apps, and it looks like things scale linearly (or even sub-linearly) based on input size, with us being able to compile between 550 and 900 bytes of input code per millisecond on my machine.

**Comment by grouma** [comment]
Can we close this?

cc @annagrin

**Comment by jakemac53** [comment]
I do thing we should still do this, it just depends on the prioritization?

**Comment by annagrin** [comment]
Agreed with @jakemac53  - we might want to come back to it in the future.
